overlord/ifacestate/handlers.go

```diff
@@ -120,11 +119,9 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, tomb *tomb.Tomb) er
 				return fmt.Errorf("cannot finish core installation, there was a rollback across reboot")
 			}
 		}
-	}
 
-	// Compatibility with old snapd: check if we have auto-connect task and if not, inject it after self (setup-profiles).
-	// In the older snapd versions interfaces were auto-connected as part of setupProfilesForSnap.
-	if snapInfo.Type != snap.TypeOS || corePhase2 {
+		// Compatibility with old snapd: check if we have auto-connect task and if not, inject it after self (setup-profiles).
```

Comment:
inject it for core after the 2nd setup-profiles, same placement as done in doInstall